File: interview-questions/leetcode977.py
#!/usr/bin/env python3
from typing import List
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        head = 0
        tail = len(nums)-1
        new_nums = [0 for _ in range(len(nums))]
        for i in range(len(nums)):
            if abs(nums[tail]) > abs(nums[head]):
                new_nums[i] = nums[tail] ** 2
                tail -= 1
            else:
                new_nums[i] = nums[head] ** 2
                head += 1
        return new_nums

File: interview-questions/leetcode977.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortedSquares = function(nums) {
    var head = 0;
    var tail = nums.length-1;
    var new_nums = new Array(nums.length);
    for(var i = nums.length-1; i >= 0; i--){
        if(Math.abs(nums[tail]) > Math.abs(nums[head])){
            new_nums[i] = nums[tail] * nums[tail];
            tail--;
        }
        else{
            new_nums[i] = nums[head] * nums[head];
            head++;
        }
    }
    return new_nums;
};
